nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/DeduplicateRecords.java

```diff
@@ -0,0 +1,690 @@
+                selectedFields.forEach(fieldVal -> {
+                    fieldVariables.clear();
+                    fieldVariables.put(FIELD_NAME, fieldVal.getField().getFieldName());
+                    fieldVariables.put(FIELD_VALUE, DataTypeUtils.toString(fieldVal.getValue(), (String) null));
+                    fieldVariables.put(FIELD_TYPE, fieldVal.getField().getDataType().getFieldType().name());
+
+                    fieldValues.add(recordPathPropertyValue.evaluateAttributeExpressions(flowFile, fieldVariables).getValue());
+                });
+            } else {
+                fieldValues.add(recordPathPropertyValue.evaluateAttributeExpressions(flowFile).getValue());
+            }
+
+            fieldValues.addAll(selectedFields.stream()
+                    .map(f -> recordPathPropertyValue.evaluateAttributeExpressions(flowFile).getValue())
+                    .collect(toList())
+            );
+        }
+
+        return Joiner.on(JOIN_CHAR).join(fieldValues);
+    }
+
+    private abstract static class FilterWrapper {
+        public static FilterWrapper create(Object filter) {
+            if(filter instanceof HashSet) {
+                return new HashSetFilterWrapper((HashSet<String>) filter);
+            } else {
+                return new BloomFilterWrapper((BloomFilter<String>) filter);
+            }
+        }
+        public abstract boolean contains(String value);
+        public abstract void put(String value);
+    }
+
+    private static class HashSetFilterWrapper extends FilterWrapper {
+
+        private final HashSet<String> filter;
+
+        public HashSetFilterWrapper(HashSet<String> filter) {
+            this.filter = filter;
+        }
+
+        @Override
+        public boolean contains(String value) {
+            return filter.contains(value);
+        }
+
+        @Override
+        public void put(String value) {
+            filter.add(value);
+        }
+    }
+
+    private static class BloomFilterWrapper extends FilterWrapper {
+
+        private final BloomFilter<String> filter;
+
+        public BloomFilterWrapper(BloomFilter<String> filter) {
+            this.filter = filter;
+        }
+
+        @Override
+        public boolean contains(String value) {
+            return filter.mightContain(value);
+        }
+
+        @Override
+        public void put(String value) {
+            filter.put(value);
+        }
+    }
+
+    private static class DistributedMapCacheClientWrapper extends FilterWrapper {
+        private DistributedMapCacheClient client;
+
+        public DistributedMapCacheClientWrapper(DistributedMapCacheClient client) {
+            this.client = client;
+        }
+
+        @Override
+        public boolean contains(String value) {
+            try {
+                return client.containsKey(value, STRING_SERIALIZER);
+            } catch (IOException e) {
+                throw new ProcessException(e);
```

Comment:
It would be helpful to provide a message instead of just wrapping the exception:
```suggestion
                throw new ProcessException("Distributed Map lookup failed", e);
```